lib/AnsiSerializer.js

```diff
@@ -127,7 +128,7 @@ AnsiSerializer.prototype.text = function (options) {
 
                 var open = ansiStyles[styleName].open;
                 var close = ansiStyles[styleName].close;
-                if (color16Hex !== color256Hex) {
+                if (this.supportsColor.has256) {
```

Comment:
We can avoid outputting both the 256 color and 16 color preambles here if the terminal supports it (can be fixed subsequently).